<html>
<head>
<title>Include/pyerrors.h</title>
<meta name='robots' content='noindex,nofollow'>
<meta name='generator' content='GLOBAL-5.8.1'>
</head>
<body text='#191970' bgcolor='#f5f5dc' vlink='gray'>
<a name='TOP'><h2><a href='../mains.html'>root</a>/<a href='../files/3349.html'>Include</a>/pyerrors.h</h2>
<i><font color='green'>/* [&lt;][&gt;][^][v][top]<a href='#BOTTOM'>[bottom]</a><a href='../mains.html'>[index]</a><a href='../help.html'>[help]</a> */</font></i>
<hr>
<h2><a href='570.html#L106' title='Included from 106 in Include/Python.h.'>INCLUDED FROM</a></h2>
<hr>
<pre>
<a name='L1'><font color='darkred'>#ifndef</font> <a href='../S/558.html#L2' title='Defined at 2 in Include/pyerrors.h.'>Py_ERRORS_H</a>
<a name='L2'><font color='darkred'>#define</font> <a href='../S/558.html#L1' title='Refered from 1 in Include/pyerrors.h.'>Py_ERRORS_H</a>
<a name='L3'><font color='darkred'>#ifdef</font> __cplusplus
<a name='L4'><b>extern</b> "C" <font color='red'>{</font>
<a name='L5'><font color='darkred'>#endif</font>
<a name='L6'>
<a name='L7'><i><font color='green'>/* Error objects */</font></i>
<a name='L8'>
<a name='L9'><font color='darkred'>#ifndef</font> Py_LIMITED_API
<a name='L10'><i><font color='green'>/* PyException_HEAD defines the initial segment of every exception class. */</font></i>
<a name='L11'><font color='darkred'>#define</font> <a href='../R/2930.html' title='Multiple refered from 7 places.'>PyException_HEAD</a> <a href='../S/544.html#L82' title='Defined at 82 in Include/object.h.'>PyObject_HEAD</a> <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *dict;\
<a name='L12'>             <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *<a href='../S/2858.html#L294' title='Defined at 294 in PC/bdist_wininst/install.c.'>args</a>; <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *traceback;\
<a name='L13'>             <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *context; <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *cause;
<a name='L14'>
<a name='L15'><b>typedef</b> <b>struct</b> <font color='red'>{</font>
<a name='L16'>    <a href='../S/558.html#L11' title='Defined at 11 in Include/pyerrors.h.'>PyException_HEAD</a>
<a name='L17'><font color='red'>}</font> <a href='../R/2508.html' title='Multiple refered from 42 places.'>PyBaseExceptionObject</a>;
<a name='L18'>
<a name='L19'><b>typedef</b> <b>struct</b> <font color='red'>{</font>
<a name='L20'>    <a href='../S/558.html#L11' title='Defined at 11 in Include/pyerrors.h.'>PyException_HEAD</a>
<a name='L21'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *msg;
<a name='L22'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *filename;
<a name='L23'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *lineno;
<a name='L24'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *offset;
<a name='L25'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *text;
<a name='L26'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *print_file_and_line;
<a name='L27'><font color='red'>}</font> <a href='../R/3414.html' title='Multiple refered from 6 places.'>PySyntaxErrorObject</a>;
<a name='L28'>
<a name='L29'><b>typedef</b> <b>struct</b> <font color='red'>{</font>
<a name='L30'>    <a href='../S/558.html#L11' title='Defined at 11 in Include/pyerrors.h.'>PyException_HEAD</a>
<a name='L31'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *<a href='../D/8346.html' title='Multiple defined in 2 places.'>encoding</a>;
<a name='L32'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *<a href='../S/500.html#L6' title='Defined at 6 in Include/asdl.h.'>object</a>;
<a name='L33'>    <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> start;
<a name='L34'>    <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> end;
<a name='L35'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *reason;
<a name='L36'><font color='red'>}</font> <a href='../R/3528.html' title='Multiple refered from 44 places.'>PyUnicodeErrorObject</a>;
<a name='L37'>
<a name='L38'><b>typedef</b> <b>struct</b> <font color='red'>{</font>
<a name='L39'>    <a href='../S/558.html#L11' title='Defined at 11 in Include/pyerrors.h.'>PyException_HEAD</a>
<a name='L40'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *<a href='../S/2755.html#L28' title='Defined at 28 in Modules/zlib/inftrees.h.'>code</a>;
<a name='L41'><font color='red'>}</font> <a href='../S/2789.html#L531' title='Refered from 531 in Objects/exceptions.c.'>PySystemExitObject</a>;
<a name='L42'>
<a name='L43'><b>typedef</b> <b>struct</b> <font color='red'>{</font>
<a name='L44'>    <a href='../S/558.html#L11' title='Defined at 11 in Include/pyerrors.h.'>PyException_HEAD</a>
<a name='L45'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *myerrno;
<a name='L46'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *strerror;
<a name='L47'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *filename;
<a name='L48'><font color='red'>}</font> <a href='../R/2863.html' title='Multiple refered from 5 places.'>PyEnvironmentErrorObject</a>;
<a name='L49'>
<a name='L50'><font color='darkred'>#ifdef</font> <a href='../S/2906.html#L81' title='Defined at 81 in PC/pyconfig.h.'>MS_WINDOWS</a>
<a name='L51'><b>typedef</b> <b>struct</b> <font color='red'>{</font>
<a name='L52'>    <a href='../S/558.html#L11' title='Defined at 11 in Include/pyerrors.h.'>PyException_HEAD</a>
<a name='L53'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *myerrno;
<a name='L54'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *strerror;
<a name='L55'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *filename;
<a name='L56'>    <a href='../D/3593.html' title='Multiple defined in 2 places.'>PyObject</a> *winerror;
<a name='L57'><font color='red'>}</font> <a href='../R/3624.html' title='Multiple refered from 4 places.'>PyWindowsErrorObject</a>;
<a name='L58'><font color='darkred'>#endif</font>
<a name='L59'><font color='darkred'>#endif</font>
<a name='L60'>
<a name='L61'><i><font color='green'>/* Error handling definitions */</font></i>
<a name='L62'>
<a name='L63'><a href='../D/2716.html' title='Multiple defined in 45 places.'>PyAPI_FUNC</a>(<b>void</b>) PyErr_SetNone(PyObject *);
<a name='L64'>PyAPI_FUNC(<b>void</b>) PyErr_SetObject(PyObject *, PyObject *);
<a name='L65'>PyAPI_FUNC(<b>void</b>) PyErr_SetString(
<a name='L66'>    PyObject *exception,
<a name='L67'>    <b>const</b> <b>char</b> *string   <i><font color='green'>/* decoded from utf-8 */</font></i>
<a name='L68'>    );
<a name='L69'>PyAPI_FUNC(PyObject *) PyErr_Occurred(<b>void</b>);
<a name='L70'>PyAPI_FUNC(<b>void</b>) PyErr_Clear(<b>void</b>);
<a name='L71'>PyAPI_FUNC(<b>void</b>) PyErr_Fetch(PyObject **, PyObject **, PyObject **);
<a name='L72'>PyAPI_FUNC(<b>void</b>) PyErr_Restore(PyObject *, PyObject *, PyObject *);
<a name='L73'>PyAPI_FUNC(<b>void</b>) Py_FatalError(<b>const</b> <b>char</b> *message);
<a name='L74'>
<a name='L75'><font color='darkred'>#if</font> <b>defined</b>(<a href='../S/2906.html#L366' title='Defined at 366 in PC/pyconfig.h.'>Py_DEBUG</a>) || <b>defined</b>(Py_LIMITED_API)
<a name='L76'><font color='darkred'>#define</font> _PyErr_OCCURRED() PyErr_Occurred()
<a name='L77'><font color='darkred'>#else</font>
<a name='L78'><font color='darkred'>#define</font> _PyErr_OCCURRED() (_PyThreadState_Current-&gt;curexc_type)
<a name='L79'><font color='darkred'>#endif</font>
<a name='L80'>
<a name='L81'><i><font color='green'>/* Error testing and normalization */</font></i>
<a name='L82'>PyAPI_FUNC(<b>int</b>) PyErr_GivenExceptionMatches(PyObject *, PyObject *);
<a name='L83'>PyAPI_FUNC(<b>int</b>) PyErr_ExceptionMatches(PyObject *);
<a name='L84'>PyAPI_FUNC(<b>void</b>) PyErr_NormalizeException(PyObject**, PyObject**, PyObject**);
<a name='L85'>
<a name='L86'><i><font color='green'>/* Traceback manipulation (PEP 3134) */</font></i>
<a name='L87'>PyAPI_FUNC(<b>int</b>) PyException_SetTraceback(PyObject *, PyObject *);
<a name='L88'>PyAPI_FUNC(PyObject *) PyException_GetTraceback(PyObject *);
<a name='L89'>
<a name='L90'><i><font color='green'>/* Cause manipulation (PEP 3134) */</font></i>
<a name='L91'>PyAPI_FUNC(PyObject *) PyException_GetCause(PyObject *);
<a name='L92'>PyAPI_FUNC(<b>void</b>) PyException_SetCause(PyObject *, PyObject *);
<a name='L93'>
<a name='L94'><i><font color='green'>/* Context manipulation (PEP 3134) */</font></i>
<a name='L95'>PyAPI_FUNC(PyObject *) PyException_GetContext(PyObject *);
<a name='L96'>PyAPI_FUNC(<b>void</b>) PyException_SetContext(PyObject *, PyObject *);
<a name='L97'>
<a name='L98'>
<a name='L99'><i><font color='green'>/* */</font></i>
<a name='L100'>
<a name='L101'><font color='darkred'>#define</font> PyExceptionClass_Check(x)                                       \
<a name='L102'>    (PyType_Check((x)) &amp;&amp;                                               \
<a name='L103'>     PyType_FastSubclass((PyTypeObject*)(x), Py_TPFLAGS_BASE_EXC_SUBCLASS))
<a name='L104'>
<a name='L105'><font color='darkred'>#define</font> PyExceptionInstance_Check(x)                    \
<a name='L106'>    PyType_FastSubclass((x)-&gt;ob_type, Py_TPFLAGS_BASE_EXC_SUBCLASS)
<a name='L107'>
<a name='L108'><font color='darkred'>#define</font> PyExceptionClass_Name(x) \
<a name='L109'>     ((<b>char</b> *)(((PyTypeObject*)(x))-&gt;tp_name))
<a name='L110'>
<a name='L111'><font color='darkred'>#define</font> PyExceptionInstance_Class(x) ((PyObject*)((x)-&gt;ob_type))
<a name='L112'>
<a name='L113'>
<a name='L114'><i><font color='green'>/* Predefined exceptions */</font></i>
<a name='L115'>
<a name='L116'>PyAPI_DATA(PyObject *) PyExc_BaseException;
<a name='L117'>PyAPI_DATA(PyObject *) PyExc_Exception;
<a name='L118'>PyAPI_DATA(PyObject *) PyExc_StopIteration;
<a name='L119'>PyAPI_DATA(PyObject *) PyExc_GeneratorExit;
<a name='L120'>PyAPI_DATA(PyObject *) PyExc_ArithmeticError;
<a name='L121'>PyAPI_DATA(PyObject *) PyExc_LookupError;
<a name='L122'>
<a name='L123'>PyAPI_DATA(PyObject *) PyExc_AssertionError;
<a name='L124'>PyAPI_DATA(PyObject *) PyExc_AttributeError;
<a name='L125'>PyAPI_DATA(PyObject *) PyExc_EOFError;
<a name='L126'>PyAPI_DATA(PyObject *) PyExc_FloatingPointError;
<a name='L127'>PyAPI_DATA(PyObject *) PyExc_EnvironmentError;
<a name='L128'>PyAPI_DATA(PyObject *) PyExc_IOError;
<a name='L129'>PyAPI_DATA(PyObject *) PyExc_OSError;
<a name='L130'>PyAPI_DATA(PyObject *) PyExc_ImportError;
<a name='L131'>PyAPI_DATA(PyObject *) PyExc_IndexError;
<a name='L132'>PyAPI_DATA(PyObject *) PyExc_KeyError;
<a name='L133'>PyAPI_DATA(PyObject *) PyExc_KeyboardInterrupt;
<a name='L134'>PyAPI_DATA(PyObject *) PyExc_MemoryError;
<a name='L135'>PyAPI_DATA(PyObject *) PyExc_NameError;
<a name='L136'>PyAPI_DATA(PyObject *) PyExc_OverflowError;
<a name='L137'>PyAPI_DATA(PyObject *) PyExc_RuntimeError;
<a name='L138'>PyAPI_DATA(PyObject *) PyExc_NotImplementedError;
<a name='L139'>PyAPI_DATA(PyObject *) PyExc_SyntaxError;
<a name='L140'>PyAPI_DATA(PyObject *) PyExc_IndentationError;
<a name='L141'>PyAPI_DATA(PyObject *) PyExc_TabError;
<a name='L142'>PyAPI_DATA(PyObject *) PyExc_ReferenceError;
<a name='L143'>PyAPI_DATA(PyObject *) PyExc_SystemError;
<a name='L144'>PyAPI_DATA(PyObject *) PyExc_SystemExit;
<a name='L145'>PyAPI_DATA(PyObject *) PyExc_TypeError;
<a name='L146'>PyAPI_DATA(PyObject *) PyExc_UnboundLocalError;
<a name='L147'>PyAPI_DATA(PyObject *) PyExc_UnicodeError;
<a name='L148'>PyAPI_DATA(PyObject *) PyExc_UnicodeEncodeError;
<a name='L149'>PyAPI_DATA(PyObject *) PyExc_UnicodeDecodeError;
<a name='L150'>PyAPI_DATA(PyObject *) PyExc_UnicodeTranslateError;
<a name='L151'>PyAPI_DATA(PyObject *) PyExc_ValueError;
<a name='L152'>PyAPI_DATA(PyObject *) PyExc_ZeroDivisionError;
<a name='L153'><font color='darkred'>#ifdef</font> <a href='../S/2906.html#L81' title='Defined at 81 in PC/pyconfig.h.'>MS_WINDOWS</a>
<a name='L154'>PyAPI_DATA(PyObject *) PyExc_WindowsError;
<a name='L155'><font color='darkred'>#endif</font>
<a name='L156'><font color='darkred'>#ifdef</font> __VMS
<a name='L157'>PyAPI_DATA(PyObject *) PyExc_VMSError;
<a name='L158'><font color='darkred'>#endif</font>
<a name='L159'>
<a name='L160'>PyAPI_DATA(PyObject *) PyExc_BufferError;
<a name='L161'>
<a name='L162'>PyAPI_DATA(PyObject *) PyExc_RecursionErrorInst;
<a name='L163'>
<a name='L164'><i><font color='green'>/* Predefined warning categories */</font></i>
<a name='L165'>PyAPI_DATA(PyObject *) PyExc_Warning;
<a name='L166'>PyAPI_DATA(PyObject *) PyExc_UserWarning;
<a name='L167'>PyAPI_DATA(PyObject *) PyExc_DeprecationWarning;
<a name='L168'>PyAPI_DATA(PyObject *) PyExc_PendingDeprecationWarning;
<a name='L169'>PyAPI_DATA(PyObject *) PyExc_SyntaxWarning;
<a name='L170'>PyAPI_DATA(PyObject *) PyExc_RuntimeWarning;
<a name='L171'>PyAPI_DATA(PyObject *) PyExc_FutureWarning;
<a name='L172'>PyAPI_DATA(PyObject *) PyExc_ImportWarning;
<a name='L173'>PyAPI_DATA(PyObject *) PyExc_UnicodeWarning;
<a name='L174'>PyAPI_DATA(PyObject *) PyExc_BytesWarning;
<a name='L175'>PyAPI_DATA(PyObject *) PyExc_ResourceWarning;
<a name='L176'>
<a name='L177'>
<a name='L178'><i><font color='green'>/* Convenience functions */</font></i>
<a name='L179'>
<a name='L180'>PyAPI_FUNC(<b>int</b>) PyErr_BadArgument(<b>void</b>);
<a name='L181'>PyAPI_FUNC(PyObject *) PyErr_NoMemory(<b>void</b>);
<a name='L182'>PyAPI_FUNC(PyObject *) PyErr_SetFromErrno(PyObject *);
<a name='L183'>PyAPI_FUNC(PyObject *) PyErr_SetFromErrnoWithFilenameObject(
<a name='L184'>    PyObject *, PyObject *);
<a name='L185'>PyAPI_FUNC(PyObject *) PyErr_SetFromErrnoWithFilename(
<a name='L186'>    PyObject *exc,
<a name='L187'>    <b>const</b> <b>char</b> *filename   <i><font color='green'>/* decoded from the filesystem encoding */</font></i>
<a name='L188'>    );
<a name='L189'><font color='darkred'>#if</font> <b>defined</b>(<a href='../S/2906.html#L81' title='Defined at 81 in PC/pyconfig.h.'>MS_WINDOWS</a>) &amp;&amp; !<b>defined</b>(Py_LIMITED_API)
<a name='L190'>PyAPI_FUNC(PyObject *) PyErr_SetFromErrnoWithUnicodeFilename(
<a name='L191'>    PyObject *, <b>const</b> Py_UNICODE *);
<a name='L192'><font color='darkred'>#endif</font> <i><font color='green'>/* MS_WINDOWS */</font></i>
<a name='L193'>
<a name='L194'>PyAPI_FUNC(PyObject *) PyErr_Format(
<a name='L195'>    PyObject *exception,
<a name='L196'>    <b>const</b> <b>char</b> *format,   <i><font color='green'>/* ASCII-encoded string  */</font></i>
<a name='L197'>    ...
<a name='L198'>    );
<a name='L199'>
<a name='L200'><font color='darkred'>#ifdef</font> <a href='../S/2906.html#L81' title='Defined at 81 in PC/pyconfig.h.'>MS_WINDOWS</a>
<a name='L201'>PyAPI_FUNC(PyObject *) PyErr_SetFromWindowsErrWithFilenameObject(
<a name='L202'>    <b>int</b>, <b>const</b> <b>char</b> *);
<a name='L203'>PyAPI_FUNC(PyObject *) PyErr_SetFromWindowsErrWithFilename(
<a name='L204'>    <b>int</b> ierr,
<a name='L205'>    <b>const</b> <b>char</b> *filename        <i><font color='green'>/* decoded from the filesystem encoding */</font></i>
<a name='L206'>    );
<a name='L207'><font color='darkred'>#ifndef</font> Py_LIMITED_API
<a name='L208'><i><font color='green'>/* XXX redeclare to use WSTRING */</font></i>
<a name='L209'>PyAPI_FUNC(PyObject *) PyErr_SetFromWindowsErrWithUnicodeFilename(
<a name='L210'>    <b>int</b>, <b>const</b> Py_UNICODE *);
<a name='L211'><font color='darkred'>#endif</font>
<a name='L212'>PyAPI_FUNC(PyObject *) PyErr_SetFromWindowsErr(<b>int</b>);
<a name='L213'>PyAPI_FUNC(PyObject *) PyErr_SetExcFromWindowsErrWithFilenameObject(
<a name='L214'>    PyObject *,<b>int</b>, PyObject *);
<a name='L215'>PyAPI_FUNC(PyObject *) PyErr_SetExcFromWindowsErrWithFilename(
<a name='L216'>    PyObject *exc,
<a name='L217'>    <b>int</b> ierr,
<a name='L218'>    <b>const</b> <b>char</b> *filename        <i><font color='green'>/* decoded from the filesystem encoding */</font></i>
<a name='L219'>    );
<a name='L220'><font color='darkred'>#ifndef</font> Py_LIMITED_API
<a name='L221'>PyAPI_FUNC(PyObject *) PyErr_SetExcFromWindowsErrWithUnicodeFilename(
<a name='L222'>    PyObject *,<b>int</b>, <b>const</b> Py_UNICODE *);
<a name='L223'><font color='darkred'>#endif</font>
<a name='L224'>PyAPI_FUNC(PyObject *) PyErr_SetExcFromWindowsErr(PyObject *, <b>int</b>);
<a name='L225'><font color='darkred'>#endif</font> <i><font color='green'>/* MS_WINDOWS */</font></i>
<a name='L226'>
<a name='L227'><i><font color='green'>/* Export the old function so that the existing API remains available: */</font></i>
<a name='L228'>PyAPI_FUNC(<b>void</b>) PyErr_BadInternalCall(<b>void</b>);
<a name='L229'>PyAPI_FUNC(<b>void</b>) _PyErr_BadInternalCall(<b>const</b> <b>char</b> *filename, <b>int</b> lineno);
<a name='L230'><i><font color='green'>/* Mask the old API with a call to the new API for code compiled under</font></i>
<a name='L231'><i><font color='green'>   Python 2.0: */</font></i>
<a name='L232'><font color='darkred'>#define</font> PyErr_BadInternalCall() _PyErr_BadInternalCall(__FILE__, __LINE__)
<a name='L233'>
<a name='L234'><i><font color='green'>/* Function to create a new exception */</font></i>
<a name='L235'>PyAPI_FUNC(PyObject *) PyErr_NewException(
<a name='L236'>    <b>const</b> <b>char</b> *name, PyObject *base, PyObject *dict);
<a name='L237'>PyAPI_FUNC(PyObject *) PyErr_NewExceptionWithDoc(
<a name='L238'>    <b>const</b> <b>char</b> *name, <b>const</b> <b>char</b> *doc, PyObject *base, PyObject *dict);
<a name='L239'>PyAPI_FUNC(<b>void</b>) PyErr_WriteUnraisable(PyObject *);
<a name='L240'>
<a name='L241'><i><font color='green'>/* In sigcheck.c or signalmodule.c */</font></i>
<a name='L242'>PyAPI_FUNC(<b>int</b>) PyErr_CheckSignals(<b>void</b>);
<a name='L243'>PyAPI_FUNC(<b>void</b>) PyErr_SetInterrupt(<b>void</b>);
<a name='L244'>
<a name='L245'><i><font color='green'>/* In signalmodule.c */</font></i>
<a name='L246'><font color='darkred'>#ifndef</font> Py_LIMITED_API
<a name='L247'><b>int</b> PySignal_SetWakeupFd(<b>int</b> fd);
<a name='L248'><font color='darkred'>#endif</font>
<a name='L249'>
<a name='L250'><i><font color='green'>/* Support for adding program text to SyntaxErrors */</font></i>
<a name='L251'>PyAPI_FUNC(<b>void</b>) PyErr_SyntaxLocation(
<a name='L252'>    <b>const</b> <b>char</b> *filename,       <i><font color='green'>/* decoded from the filesystem encoding */</font></i>
<a name='L253'>    <b>int</b> lineno);
<a name='L254'>PyAPI_FUNC(<b>void</b>) PyErr_SyntaxLocationEx(
<a name='L255'>    <b>const</b> <b>char</b> *filename,       <i><font color='green'>/* decoded from the filesystem encoding */</font></i>
<a name='L256'>    <b>int</b> lineno,
<a name='L257'>    <b>int</b> col_offset);
<a name='L258'>PyAPI_FUNC(PyObject *) PyErr_ProgramText(
<a name='L259'>    <b>const</b> <b>char</b> *filename,       <i><font color='green'>/* decoded from the filesystem encoding */</font></i>
<a name='L260'>    <b>int</b> lineno);
<a name='L261'>
<a name='L262'><i><font color='green'>/* The following functions are used to create and modify unicode</font></i>
<a name='L263'><i><font color='green'>   exceptions from C */</font></i>
<a name='L264'>
<a name='L265'><i><font color='green'>/* create a UnicodeDecodeError object */</font></i>
<a name='L266'>PyAPI_FUNC(PyObject *) PyUnicodeDecodeError_Create(
<a name='L267'>    <b>const</b> <b>char</b> *encoding,       <i><font color='green'>/* UTF-8 encoded string */</font></i>
<a name='L268'>    <b>const</b> <b>char</b> *object,
<a name='L269'>    <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> length,
<a name='L270'>    <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> start,
<a name='L271'>    <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> end,
<a name='L272'>    <b>const</b> <b>char</b> *reason          <i><font color='green'>/* UTF-8 encoded string */</font></i>
<a name='L273'>    );
<a name='L274'>
<a name='L275'><i><font color='green'>/* create a UnicodeEncodeError object */</font></i>
<a name='L276'><font color='darkred'>#ifndef</font> Py_LIMITED_API
<a name='L277'>PyAPI_FUNC(PyObject *) PyUnicodeEncodeError_Create(
<a name='L278'>    <b>const</b> <b>char</b> *encoding,       <i><font color='green'>/* UTF-8 encoded string */</font></i>
<a name='L279'>    <b>const</b> Py_UNICODE *object,
<a name='L280'>    <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> length,
<a name='L281'>    <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> start,
<a name='L282'>    <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> end,
<a name='L283'>    <b>const</b> <b>char</b> *reason          <i><font color='green'>/* UTF-8 encoded string */</font></i>
<a name='L284'>    );
<a name='L285'><font color='darkred'>#endif</font>
<a name='L286'>
<a name='L287'><i><font color='green'>/* create a UnicodeTranslateError object */</font></i>
<a name='L288'><font color='darkred'>#ifndef</font> Py_LIMITED_API
<a name='L289'>PyAPI_FUNC(PyObject *) PyUnicodeTranslateError_Create(
<a name='L290'>    <b>const</b> Py_UNICODE *object,
<a name='L291'>    <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> length,
<a name='L292'>    <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> start,
<a name='L293'>    <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> end,
<a name='L294'>    <b>const</b> <b>char</b> *reason          <i><font color='green'>/* UTF-8 encoded string */</font></i>
<a name='L295'>    );
<a name='L296'><font color='darkred'>#endif</font>
<a name='L297'>
<a name='L298'><i><font color='green'>/* get the encoding attribute */</font></i>
<a name='L299'>PyAPI_FUNC(PyObject *) PyUnicodeEncodeError_GetEncoding(PyObject *);
<a name='L300'>PyAPI_FUNC(PyObject *) PyUnicodeDecodeError_GetEncoding(PyObject *);
<a name='L301'>
<a name='L302'><i><font color='green'>/* get the object attribute */</font></i>
<a name='L303'>PyAPI_FUNC(PyObject *) PyUnicodeEncodeError_GetObject(PyObject *);
<a name='L304'>PyAPI_FUNC(PyObject *) PyUnicodeDecodeError_GetObject(PyObject *);
<a name='L305'>PyAPI_FUNC(PyObject *) PyUnicodeTranslateError_GetObject(PyObject *);
<a name='L306'>
<a name='L307'><i><font color='green'>/* get the value of the start attribute (the int * may not be NULL)</font></i>
<a name='L308'><i><font color='green'>   return 0 on success, -1 on failure */</font></i>
<a name='L309'>PyAPI_FUNC(<b>int</b>) PyUnicodeEncodeError_GetStart(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> *);
<a name='L310'>PyAPI_FUNC(<b>int</b>) PyUnicodeDecodeError_GetStart(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> *);
<a name='L311'>PyAPI_FUNC(<b>int</b>) PyUnicodeTranslateError_GetStart(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> *);
<a name='L312'>
<a name='L313'><i><font color='green'>/* assign a new value to the start attribute</font></i>
<a name='L314'><i><font color='green'>   return 0 on success, -1 on failure */</font></i>
<a name='L315'>PyAPI_FUNC(<b>int</b>) PyUnicodeEncodeError_SetStart(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a>);
<a name='L316'>PyAPI_FUNC(<b>int</b>) PyUnicodeDecodeError_SetStart(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a>);
<a name='L317'>PyAPI_FUNC(<b>int</b>) PyUnicodeTranslateError_SetStart(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a>);
<a name='L318'>
<a name='L319'><i><font color='green'>/* get the value of the end attribute (the int *may not be NULL)</font></i>
<a name='L320'><i><font color='green'> return 0 on success, -1 on failure */</font></i>
<a name='L321'>PyAPI_FUNC(<b>int</b>) PyUnicodeEncodeError_GetEnd(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> *);
<a name='L322'>PyAPI_FUNC(<b>int</b>) PyUnicodeDecodeError_GetEnd(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> *);
<a name='L323'>PyAPI_FUNC(<b>int</b>) PyUnicodeTranslateError_GetEnd(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a> *);
<a name='L324'>
<a name='L325'><i><font color='green'>/* assign a new value to the end attribute</font></i>
<a name='L326'><i><font color='green'>   return 0 on success, -1 on failure */</font></i>
<a name='L327'>PyAPI_FUNC(<b>int</b>) PyUnicodeEncodeError_SetEnd(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a>);
<a name='L328'>PyAPI_FUNC(<b>int</b>) PyUnicodeDecodeError_SetEnd(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a>);
<a name='L329'>PyAPI_FUNC(<b>int</b>) PyUnicodeTranslateError_SetEnd(PyObject *, <a href='../D/4364.html' title='Multiple defined in 6 places.'>Py_ssize_t</a>);
<a name='L330'>
<a name='L331'><i><font color='green'>/* get the value of the reason attribute */</font></i>
<a name='L332'>PyAPI_FUNC(PyObject *) PyUnicodeEncodeError_GetReason(PyObject *);
<a name='L333'>PyAPI_FUNC(PyObject *) PyUnicodeDecodeError_GetReason(PyObject *);
<a name='L334'>PyAPI_FUNC(PyObject *) PyUnicodeTranslateError_GetReason(PyObject *);
<a name='L335'>
<a name='L336'><i><font color='green'>/* assign a new value to the reason attribute</font></i>
<a name='L337'><i><font color='green'>   return 0 on success, -1 on failure */</font></i>
<a name='L338'>PyAPI_FUNC(<b>int</b>) PyUnicodeEncodeError_SetReason(
<a name='L339'>    PyObject *exc,
<a name='L340'>    <b>const</b> <b>char</b> *reason          <i><font color='green'>/* UTF-8 encoded string */</font></i>
<a name='L341'>    );
<a name='L342'>PyAPI_FUNC(<b>int</b>) PyUnicodeDecodeError_SetReason(
<a name='L343'>    PyObject *exc,
<a name='L344'>    <b>const</b> <b>char</b> *reason          <i><font color='green'>/* UTF-8 encoded string */</font></i>
<a name='L345'>    );
<a name='L346'>PyAPI_FUNC(<b>int</b>) PyUnicodeTranslateError_SetReason(
<a name='L347'>    PyObject *exc,
<a name='L348'>    <b>const</b> <b>char</b> *reason          <i><font color='green'>/* UTF-8 encoded string */</font></i>
<a name='L349'>    );
<a name='L350'>
<a name='L351'>
<a name='L352'><i><font color='green'>/* These APIs aren't really part of the error implementation, but</font></i>
<a name='L353'><i><font color='green'>   often needed to format error messages; the native C lib APIs are</font></i>
<a name='L354'><i><font color='green'>   not available on all platforms, which is why we provide emulations</font></i>
<a name='L355'><i><font color='green'>   for those platforms in Python/mysnprintf.c,</font></i>
<a name='L356'><i><font color='green'>   WARNING:  The return value of snprintf varies across platforms; do</font></i>
<a name='L357'><i><font color='green'>   not rely on any particular behavior; eventually the C99 defn may</font></i>
<a name='L358'><i><font color='green'>   be reliable.</font></i>
<a name='L359'><i><font color='green'>*/</font></i>
<a name='L360'><font color='darkred'>#if</font> <b>defined</b>(<a href='../S/2906.html#L80' title='Defined at 80 in PC/pyconfig.h.'>MS_WIN32</a>) &amp;&amp; !<b>defined</b>(<a href='../S/2894.html#L327' title='Defined at 327 in PC/os2emx/pyconfig.h.'>HAVE_SNPRINTF</a>)
<a name='L361'><font color='darkred'># define</font> <a href='../S/2894.html#L327' title='Defined at 327 in PC/os2emx/pyconfig.h.'>HAVE_SNPRINTF</a>
<a name='L362'><font color='darkred'># define</font> snprintf _snprintf
<a name='L363'><font color='darkred'># define</font> vsnprintf _vsnprintf
<a name='L364'><font color='darkred'>#endif</font>
<a name='L365'>
<a name='L366'><font color='darkred'>#include</font> &lt;stdarg.h&gt;
<a name='L367'>PyAPI_FUNC(<b>int</b>) PyOS_snprintf(<b>char</b> *str, <a href='../D/11182.html' title='Multiple defined in 2 places.'>size_t</a> size, <b>const</b> <b>char</b>  *format, ...)
<a name='L368'>                        Py_GCC_ATTRIBUTE((format(printf, 3, 4)));
<a name='L369'>PyAPI_FUNC(<b>int</b>) PyOS_vsnprintf(<b>char</b> *str, <a href='../D/11182.html' title='Multiple defined in 2 places.'>size_t</a> size, <b>const</b> <b>char</b>  *format, va_list va)
<a name='L370'>                        Py_GCC_ATTRIBUTE((format(printf, 3, 0)));
<a name='L371'>
<a name='L372'><font color='darkred'>#ifdef</font> __cplusplus
<a name='L373'><font color='red'>}</font>
<a name='L374'><font color='darkred'>#endif</font>
<a name='L375'><font color='darkred'>#endif</font> <i><font color='green'>/* !Py_ERRORS_H */</font></i>
</pre>
<hr>
<a name='BOTTOM'>
<i><font color='green'>/* [&lt;][&gt;][^][v]<a href='#TOP'>[top]</a>[bottom]<a href='../mains.html'>[index]</a><a href='../help.html'>[help]</a> */</font></i>
</body>
</html>
